BDArmory/FX/ExplosionFX.cs

```diff
@@ -972,7 +974,14 @@ private void ExecutePartBlastEvent(PartBlastHitEvent eventToExecute)
                                     //else ProjectileUtils.CalculateArmorDamage(part, penetrationFactor, Caliber, hardness, Ductility, Density, ExplosionVelocity, SourceVesselName, ExplosionSourceType.Missile, type);
                                     else if (penetrationFactor > 0)
                                     {
-                                        ProjectileUtils.CalculateArmorDamage(part, penetrationFactor, Caliber * 2.5f, hardness, Ductility, Density, warheadType == WarheadTypes.ShapedCharge ? 5000f : ExplosionVelocity, SourceVesselName, ExplosionSourceType.Missile, type);
+                                        ProjectileUtils.CalculateArmorDamage(part, penetrationFactor, Caliber * 2.5f, hardness, Ductility, Density,
+                                            warheadType switch
+                                            {
+                                                WarheadTypes.ShapedCharge => 5000f,
+                                                WarheadTypes.Kinetic => ImpactSpeed,
+                                                _ => ExplosionVelocity
+                                            },
+                                            SourceVesselName, ExplosionSourceType.Missile, type);
```

Comment:
a couple lines down on 996 ...warheadType == WarheadTypes.ShapedCharge ? 5000f : ExplosionVelocity also needs to become WarheadType == WarheadTypes.Kinetic ? ImpactSpeed : Explosion Velocity